File: pkgs/sdks/python/examples/example_usage.py
from eunomia_core import enums, schemas
from eunomia_sdk.client import EunomiaClient


def run():
    client = EunomiaClient(endpoint="http://localhost:8000")

    try:
        # Register a principal
        print("Registering principal...")
        principal = client.register_entity(
            type=enums.EntityType.principal,
            attributes={
                "role": "admin",
                "department": "engineering",
            },
            uri="user:john.doe",
        )
        print(f"Principal registered: {principal.uri}")

        # Register a resource
        print("Registering resource...")
        resource = client.register_entity(
            type=enums.EntityType.resource,
            attributes={
                "type": "document",
                "classification": "confidential",
            },
            uri="document:project-plan",
        )
        print(f"Resource registered: {resource.uri}")

        # Check permissions
        print("Checking permissions...")
        is_allowed = client.check(
            principal_uri=principal.uri,
            resource_uri=resource.uri,
        )
        print(f"Is allowed: {is_allowed}")

        # Update entity
        print("Updating entity...")
        _ = client.update_entity(
            uri=resource.uri,
            attributes={
                "classification": "public",
                "status": "published",
            },
        )
        print("Resource updated")

        # Delete entity
        print("Deleting entity...")
        client.delete_entity(principal.uri)
        print("Entities deleted")

        # Create policy
        print("Creating policy...")
        request = schemas.CheckRequest(
            principal=schemas.PrincipalCheck(
                attributes={"role": "admin", "department": "engineering"}
            ),
            resource=schemas.ResourceCheck(
                attributes={"type": "document", "classification": "confidential"}
            ),
            action="read",
        )
        policy = client.create_simple_policy(request=request, name="policy-example")
        print(f"Policy created: {policy.name}")

    except Exception as error:
        print(f"Error: {error}")


if __name__ == "__main__":
    run()

File: pkgs/sdks/typescript/examples/example-usage.ts
/* eslint-disable no-console */
import { CheckRequest, EntityType, EunomiaClient } from "eunomia-sdk";

async function run(): Promise<void> {
  const client = new EunomiaClient({
    endpoint: "http://localhost:8000",
  });

  try {
    // // Register a principal
    console.log("Registering principal...");
    const principal = await client.registerEntity({
      type: EntityType.Principal,
      attributes: {
        role: "admin",
        department: "engineering",
      },
      uri: "user:john.doe",
    });
    console.log(`Principal registered: ${principal.uri}`);

    // Register a resource
    console.log("Registering resource...");
    const resource = await client.registerEntity({
      type: EntityType.Resource,
      attributes: {
        type: "document",
        classification: "confidential",
      },
      uri: "document:project-plan",
    });
    console.log(`Resource registered: ${resource.uri}`);

    // Check permissions
    console.log("Checking permissions...");
    const isAllowed = await client.check({
      principalUri: principal.uri,
      resourceUri: resource.uri,
    });
    console.log(`Is allowed: ${isAllowed}`);

    // Update entity
    console.log("Updating entity...");
    await client.updateEntity({
      uri: resource.uri,
      attributes: {
        classification: "public",
        status: "published",
      },
    });
    console.log("Resource updated");

    // Delete entity
    console.log("Deleting entity...");
    await client.deleteEntity(principal.uri);
    console.log("Entities deleted");

    // Create policy
    console.log("Creating policy...");
    const request: CheckRequest = {
      principal: {
        type: EntityType.Principal,
        attributes: {
          role: "admin",
          department: "engineering",
        },
      },
      resource: {
        type: EntityType.Resource,
        attributes: {
          type: "document",
          classification: "confidential",
        },
      },
      action: "access",
    };
    const policy = await client.createSimplePolicy(request, "policy-example");
    console.log(`Policy created: ${policy.name}`);
  } catch (error) {
    console.error("Error:", error);
  }
}

run().catch(console.error);
